reV/SAM/generation.py

```diff
@@ -1149,6 +1161,221 @@ def default():
+            required variables to run the respective SAM simulation.
+        meta : pd.DataFrame | pd.Series
+            Meta data corresponding to the resource input for the single
+            location. Should include values for latitude, longitude,
+            elevation, and timezone.
+        """
+        super().set_resource_data(resource, meta)
+        self._set_resource_temperature(resource)
+        self._set_nameplate_to_match_resource_potential(resource)
+        self._set_resource_potential_to_match_gross_output()
+
+    def _set_resource_temperature(self, resource):
+        """Set resource temp from data if user did not specify it. """
+
+        if "resource_temp" in self.sam_sys_inputs:
+            logger.debug("Found 'resource_temp' value in SAM config: {:.2f}"
+                         .format(self.sam_sys_inputs["resource_temp"]))
+            return
+
+        val = set(resource["temperature"].unique())
+        logger.debug("Found {} value(s) for 'temperature' in resource data"
+                     .format(len(val)))
+        if len(val) > 1:
+            msg = ("Found multiple values for 'temperature' for site {}: {}"
+                   .format(self.site, val))
+            logger.error(msg)
+            raise InputError(msg)
+
+        val = val.pop()
+        logger.debug("Input 'resource_temp' not found in SAM config - setting "
+                     "to {:.2f} based on input resource data."
+                     .format(val))
+        self.sam_sys_inputs["resource_temp"] = val
+
+    def _set_nameplate_to_match_resource_potential(self, resource):
+        """Set the nameplate capacity to match the resource potential. """
+
+        val = set(resource["potential_MW"].unique())
+        if len(val) > 1:
+            msg = ('Found multiple values for "potential_MW" for site {}: {}'
+                   .format(self.site, val))
+            logger.error(msg)
+            raise InputError(msg)
+
+        val = val.pop() * 1000
+        if "nameplate" in self.sam_sys_inputs:
+            msg = ('Setting "nameplate" is not allowed! Updating '
+                   'user input of {} to match the resource potential: {}'
+                   .format(self.sam_sys_inputs["nameplate"], val))
+            logger.warning(msg)
+            warn(msg)
+
+        logger.debug("Setting the nameplate to {}".format(val))
+        self.sam_sys_inputs["nameplate"] = val
+
+    def _set_resource_potential_to_match_gross_output(self):
+        """Set the resource potential input to match the gross generation.
+
+        As of 12/20/2022, the resource potential input is only used to
+        calculate the number of well replacements during the lifetime
+        of a geothermal plant. It was decided that reV would not model
+        well replacements, and therefore it is fine to set the resource
+        potential to match (or be just above) the gross potential so
+        that SAM does not throw any errors.
```

Comment:
good documentation and implementation. Consider putting this blurb in the class docstring though. Hidden methods don't get an html docs thing in the gh pages.

Reply:
Yea good point, I will move that up